File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int mx=0,a=p[0];
        for(int i=1;i<p.size();i++){
            int profit=p[i]-a;
            mx=max(mx,profit);
            a=min(p[i],a);
        }
        return mx;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] p) {
        int ans=0,cost=p[0];
        for(int i=0;i<p.length;i++){
            int profit=p[i]-cost;
            ans=Math.max(profit,ans);
            cost=Math.min(cost,p[i]);
        }
        return ans;
    }
}